src/Umbraco.Core/Persistence/Repositories/VersionableRepositoryBase.cs

```diff
@@ -372,32 +422,76 @@ private Sql GetSortedSqlForPagedResults(Sql sql, Direction orderDirection, strin
             return result;
         }
 
+        protected IDictionary<int, PropertyCollection> GetPropertyCollection(DocumentDefinition[] ddefs)
+        {
+            var versions = ddefs.Select(x => x.Version).ToArray();
+            if (versions.Length == 0) return new Dictionary<int, PropertyCollection>();
+
+            var sql = new Sql()
+                .Select<PropertyDataDto>(r => r.Select<PropertyTypeDto>())
+                .From<PropertyDataDto>(SqlSyntax)
+                .LeftJoin<PropertyTypeDto>(SqlSyntax).On<PropertyDataDto, PropertyTypeDto>(SqlSyntax, left => left.PropertyTypeId, right => right.Id)
+                .WhereIn<PropertyDataDto>(SqlSyntax, x => x.VersionId, versions);

```

Comment:
This will throw an exception if there are more than 2100 if we are going to do a WHERE IN then we need to do a loop using InGroupsOf(2000) like we do elsewhere